cdap-security/src/main/java/co/cask/cdap/security/store/FileSecureStore.java

```diff
@@ -80,7 +82,14 @@
     // Get the keystore password
     password = cConf.get(Constants.Security.Store.FILE_PASSWORD).toCharArray();
 
-    keyStore = locateKeystore(path, password);
+    try {
+      keyStore = locateKeystore(path, password);
+    } catch (IOException ioe) {
+      // Throw a runtime exception so that it can be thrown by the injector
```

Comment:
Then the get in the Provider would need to handle it but there is no course of action if the construction failed.